lib/grpc_web/client/client_executor.rb

```diff
@@ -16,28 +16,28 @@ class << self
     GRPC_MESSAGE_HEADER = 'grpc-message'
     GRPC_HEADERS = %W[x-grpc-web #{GRPC_STATUS_HEADER} #{GRPC_MESSAGE_HEADER}].freeze
 
-    def request(uri, rpc_desc, params = {})
+    def request(uri, rpc_desc, params = {}, opt_headers = {})
       req_proto = rpc_desc.input.new(params)
       marshalled_proto = rpc_desc.marshal_proc.call(req_proto)
       frame = ::GRPCWeb::MessageFrame.payload_frame(marshalled_proto)
       request_body = ::GRPCWeb::MessageFraming.pack_frames([frame])
 
-      resp = post_request(uri, request_body)
+      resp = post_request(uri, request_body, opt_headers)
       resp_body = handle_response(resp)
       rpc_desc.unmarshal_proc(:output).call(resp_body)
     end
 
     private
 
-    def request_headers
+    def request_headers(opt_headers)
       {
         'Accept' => PROTO_CONTENT_TYPE,
         'Content-Type' => PROTO_CONTENT_TYPE,
-      }
+      }.merge(opt_headers)
```

Comment:
For consistency with grpc-ruby, the headers should be nested under the key `:metadata`.
```suggestion
      }.merge(opt_headers[:metadata])
```

Feedback based on this SO answer
https://stackoverflow.com/a/45859842

Doubled checked it's still true in grpc-ruby specs:
https://github.com/grpc/grpc/blob/master/src/ruby/spec/generic/rpc_server_spec.rb#L386